### ISSUE TITLE
cannot check if a phone number is on whatsapp or not - METHOD NOT IMPLEMENTED BY THE ENGINE

### ISSUE THREAD

**Issue by behind24proxies** [allburov/whatsapp-http-api#37]
hi every time that I check if a phone number is on whatsapp I get a 422
```
{
  "statusCode": 422,
  "message": "The method is not implemented by the engine.",
  "error": "Unprocessable Entity"
}
```

I tried using both engines, but with VENOM , it's worse , it returns a 500

**Comment by allburov** [comment]
>it returns a 500

Could you share the full response please?

**Comment by behind24proxies** [comment]
hi, so , yesterday it was giving me 500 
now it's working fine with VENOM.

how many number verifications/checks can I do before getting banned ?

**Comment by behind24proxies** [comment]
okay, it's coming again
![image](https://user-images.githubusercontent.com/96356684/209337564-ef0785e5-9c4f-483e-ac48-70eee4461f66.png)

**Comment by behind24proxies** [comment]
okay , after debugging I found the problem and the solution :

The issue is that when a check_phone_number request is made before the webhooks are initialized on the server, it does not return a response. In other cases, the response contains an error stating that the server cannot read the properties of the WhatsApp service and is unable to retrieve the check_number.

Upon further investigation, it was determined that when the check_phone_number request is made before the websockets are fully loaded on the server, it causes a freeze in the docker container, which prevents the server from being able to send messages afterwards. However, the server is still able to refresh the Swagger UI.

To resolve this issue, it is recommended to ensure that the webhooks are fully initialized on the server before making the endpoints available for the , in order to avoid the freeze in the docker container.

**Comment by allburov** [comment]
Hi! We've added [contacts endpoints](https://waha.devlike.pro/docs/how-to/contacts/) in WAHA

Right now it supports:
- Get all contacts or a single one
- Get "about" contact (aka status)
- Get contact's profile picture
- **Check phone number is registered in WhatsApp** (works even if the number is not in your contact list)
- Block and unblock contacts

It supports default engine, so no needs to use venom for phone number check. Please have a look, it may help you!

The endpoints are available in WAHA Plus `2023.1` version and we'll add them in Core version in February 2023

**Comment by allburov** [comment]
>To resolve this issue, it is recommended to ensure that the webhooks are fully initialized on the server before making the endpoints available for the , in order to avoid the freeze in the docker container.

Great idea, thank you! We'll work on that https://github.com/devlikeapro/whatsapp-http-api/issues/40
